parachain/integration-test/src/lib.rs

```diff
@@ -0,0 +1,417 @@
+		.tx()
+		.sign_and_submit_then_watch_default(&transfer_call, &alice_signer)
+		.await?
+		.wait_for_finalized_success()
+		.await?
+		.all_events_in_block()
+		.clone();
+
+	let tx_block_hash = result.block_hash();
+
+	let events = client_a.events().at(tx_block_hash).await?;
+	log::info!("Ismp Events: {:?} \n", events.find_last::<Request>()?);
+
+	// Assert burnt & transferred tokens in chain A
+	let alice_chain_a_new_balance = client_a
+		.storage()
+		.at_latest()
+		.await?
+		.fetch(&alice_key)
+		.await?
+		.ok_or("Failed to fetch")
+		.unwrap()
+		.data
+		.free;
+
+	tokio::time::sleep(Duration::from_secs(40)).await;
+
+	// The relayer should finish sending the request message to chain B
+
+	let alice_chain_b_new_balance = client_b
+		.storage()
+		.at_latest()
+		.await?
+		.fetch(&alice_key)
+		.await?
+		.ok_or("Failed to fetch")
+		.unwrap()
+		.data
+		.free;
+
+	// diving by 10000000000 for better assertion as the rem balance = initial - amount - fees
+	// in chain A
+	assert_eq!(
+		(alice_chain_a_initial_balance - amount) / 100000000000,
+		alice_chain_a_new_balance / 100000000000
+	);
+	// in chain B
+	assert_eq!(
+		(alice_chain_b_initial_balance + amount) / 100000000000,
+		alice_chain_b_new_balance / 100000000000
+	);
+
+	Ok(())
+}
+
+/// fetch a foreign storage item from a given key
+#[tokio::test(flavor = "multi_thread")]
+async fn get_request_works() -> Result<(), anyhow::Error> {
+	log_setup()?;
+	let (
+		(client_a, client_b),
+		(chain_a_sub_client, chain_b_sub_client),
+		(chain_a_client, chain_b_client),
+		_,
+	) = initial_setup().await?;
+
+	log::info!(
+		"🧊integration test for para: {} to para {}: get request",
+		chain_a_sub_client.state_machine_id().state_id,
+		chain_b_sub_client.state_machine_id().state_id
+	);
+
+	// Accounts & keys
+	let alice_signer = PairSigner::<Hyperbridge, _>::new(
+		Pair::from_string("//Alice", None).expect("Unable to create ALice account"),
+	);
+	// parachain info pallet fetching para id
+	let encoded_chain_b_id_storage_key =
+		"0x0d715f2646c8f85767b5d2764bb2782604a74d81251e398fd8a0a4d55023bb3f";
+
+	// Chain A fetch Alice balance from chain B
+	let latest_height_b =
+		chain_a_client.query_latest_height(chain_b_client.state_machine_id()).await?;
+
+	let get_request = api::tx().ismp_demo().get_request(GetRequest {
+		para_id: 2001,
+		height: latest_height_b,
+		timeout: 70,
+		keys: vec![hex::decode(encoded_chain_b_id_storage_key.strip_prefix("0x").unwrap()).unwrap()],
+	});
+	let tx_result = client_a
+		.tx()
+		.sign_and_submit_then_watch_default(&get_request, &alice_signer)
+		.await?
+		.wait_for_finalized_success()
+		.await?
+		.all_events_in_block()
+		.clone();
+
+	let tx_block_hash = tx_result.block_hash();
+	let tx_block_height = client_a.blocks().at(tx_block_hash).await?.number() as u64;
+	let events = client_a.events().at(tx_block_hash).await?;
+	log::info!("Ismp Events: {:?} \n", events.find_last::<Request>()?);
+
+	tokio::time::sleep(Duration::from_secs(20)).await;
```

Comment:
This sleep is not necessary, the state machine update already exists on chain a.